### PR TITLE
WIP: Support to_human_time jinja function

### PR DESCRIPTION
**Pull request by lakshmi-kannan**
## What? 

Include a ``to_human_time`` jinja function that can be used to fuzzy render time value in seconds. 
For example {{ elapsed_seconds || to_human_time }} will convert an integer (or long) 181 to string '3m1s'.

## Why?
See https://github.com/StackStorm/hubot-stackstorm/issues/123

## TODO

- [X] Refactor existing jinja filters to own files
- [X] Add a to_human_time method (** I think I'll need to_human_time_from_secs to be explicit)
- [X] Fix a bug in jinja rendering where jinja filter expressions weren't detected to be jinja renderable (See https://gist.github.com/lakshmi-kannan/0af3004652c12eaa9d5002cb0903a058, eaa7420, refactored 8c95a6d)
- [x] Unit tests
- [x] Documentation for jinja filters (https://github.com/StackStorm/st2docs/pull/228)
- [ ] Changelog

### CODE REVIEW

File: st2common/st2common/util/param.py [hunk 1]
```diff
@@ -88,16 +88,22 @@ def _process(G, name, value):
     if isinstance(value, str):
         value = to_unicode(value)
     template_ast = ENV.parse(value)
+    LOG.debug('Template ast: %s', template_ast)
     # Dependencies of the node represent jinja variables used in the template
     # We're connecting nodes with an edge for every depencency to traverse them in the right order
     # and also make sure that we don't have missing or cyclic dependencies upfront.
     dependencies = meta.find_undeclared_variables(template_ast)
+    LOG.debug('Dependencies: %s', dependencies)
     if dependencies:
```
Comment (manasdk) [inline st2common/st2common/util/param.py]: perhaps 

```
if jinja_utils.is_jinja_expression(value):
    G.add_node(name, template=value)
    if dependencies:
        for dependency in dependencies:
            G.add_edge(dependency, name)
else:
    G.add_node(name, value=value)
```

might be a clearer way to write this one.

Comment (lakshmi-kannan) [inline st2common/st2common/util/param.py]: Yeah, I agree. I was going to do that. I just wanted to fix the bug and write tests so I can refactor.

Comment (lakshmi-kannan) [inline st2common/st2common/util/param.py]: Turns out that refactoring https://gist.github.com/lakshmi-kannan/24fcbf5bcd05b81b765e6665af317ada breaks the tests :/.

**Comment by lakshmi-kannan** [comment]
This travis just hates me. Let me close this PR and open a new one to see if Travis assigns a new container. Can't repro this issue locally or other travis builds.

**Pull request closed without merging**